lib/core/services/networking_service.dart

```diff
@@ -1,15 +1,46 @@
+// FLUTTER / DART / THIRD-PARTIES
 import 'dart:async';
-
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter/material.dart';
 import 'package:connectivity/connectivity.dart';
 
 class NetworkingService {
   final Connectivity _connectivity = Connectivity();
 
+  // Offline mode snackbar
+  static bool _isSnackbarDismissed = false;
+  static bool _isSnackbarActive = false;
+
   Future<bool> hasConnectivity() async {
     final connectionStatus = await _connectivity.checkConnectivity();
     return connectionStatus != ConnectivityResult.none;
   }
 
+  Future displayOfflineMode(BuildContext context, AppIntl intl) async {
+    if (!await hasConnectivity()) {
+      if (!_isSnackbarActive && !_isSnackbarDismissed) {
+        final snackBar = SnackBar(
+          content: Text(intl.no_connectivity),
+          duration: const Duration(days: 365),
+          action: SnackBarAction(
+            label: intl.close_no_connectivity_snackbar.toString().toUpperCase(),
```

Comment:
Why there is a `toString()` here? `intl.close_no_connectivity_snackbar` is already a string